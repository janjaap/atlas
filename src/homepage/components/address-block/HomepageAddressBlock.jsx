import React from 'react';
import PropTypes from 'prop-types';

import HomepageBlock from '../block/HomepageBlock';

import { features } from '../../../shared/environment';

const HomepageAddressBlock = (props) => {
  const { onLinkClick } = props;
  const brkClassName = `homepage-block__item ${features.eigendommen ? '' : 'homepage-block__item--invisible'}`;

  return (
    <HomepageBlock
      onBlockLinkClick={() => onLinkClick({ dataset: 'bag', filters: {}, page: 1 })}
      title={'Adressentabel'}
      description={'Selecteer en download als spreadsheet'}
      hasTallDescription
    >
      <div className={'homepage-block__item'}>
        <button
          className={'c-link homepage-block__link'}
          title={'Bekijk Adressentabel'}
          onClick={() => onLinkClick({ dataset: 'bag', filters: {}, page: 1 })}
        >
          <span className="homepage-block__icon homepage-block__icon--bag" />
          <span className="homepage-block__label">
            Adressentabel
          </span>
        </button>
      </div>

      <div className={'homepage-block__item'}>
        <button
          className={'c-link homepage-block__link'}
          title={'Bekijk handelsregister-tabel'}
          onClick={() => onLinkClick({ dataset: 'hr', filters: {}, page: 1 })}
        >
          <span className="homepage-block__icon homepage-block__icon--hr" />
          <span className="homepage-block__label">
            Handelsregister-tabel
          </span>
        </button>
      </div>

      <div className={brkClassName}>
        <button
          className={'c-link homepage-block__link'}
          title={'Bekijk kadaster-tabel'}
          onClick={() => onLinkClick({ dataset: 'brk', filters: {}, page: 1 })}
        >
          <span className="homepage-block__icon homepage-block__icon--brk" />
          <span className="homepage-block__label">
            Kadaster-tabel
          </span>
        </button>
      </div>

    </HomepageBlock>
  );
};

HomepageAddressBlock.propTypes = {
  onLinkClick: PropTypes.func.isRequired
};

export default HomepageAddressBlock;
