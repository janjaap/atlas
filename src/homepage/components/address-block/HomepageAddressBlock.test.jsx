import React from 'react';
import { mount, shallow } from 'enzyme';

import HomepageAddressBlock from './HomepageAddressBlock';

const mockFn = jest.fn();

describe('HomepageAddressBlock', () => {
  it('should render with the default block-link', () => {
    const wrapper = shallow(
      <HomepageAddressBlock
        onLinkClick={mockFn}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle button click events', () => {
    const wrapper = mount(
      <HomepageAddressBlock
        onLinkClick={mockFn}
      />
    );

    wrapper.find('button').at(0).simulate('click');
    expect(mockFn).toHaveBeenCalledWith({ dataset: 'bag', filters: {}, page: 1 });

    wrapper.find('button').at(1).simulate('click');
    expect(mockFn).toHaveBeenCalledWith({ dataset: 'hr', filters: {}, page: 1 });

    wrapper.find('button').at(2).simulate('click');
    expect(mockFn).toHaveBeenCalledWith({ dataset: 'brk', filters: {}, page: 1 });

    wrapper.find('button').at(3).simulate('click');
    expect(mockFn).toHaveBeenCalledWith({ dataset: 'brk', filters: {}, page: 1 });
  });
});
