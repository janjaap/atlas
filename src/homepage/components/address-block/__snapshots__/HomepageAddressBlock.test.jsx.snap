// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageAddressBlock should render with the default block-link 1`] = `
<HomepageBlock
  description="Selecteer en download als spreadsheet"
  hasTallDescription={true}
  onBlockLinkClick={[Function]}
  title="Adressentabel"
>
  <div
    className="homepage-block__item"
  >
    <button
      className="c-link homepage-block__link"
      onClick={[Function]}
      title="Bekijk Adressentabel"
    >
      <span
        className="homepage-block__icon homepage-block__icon--bag"
      />
      <span
        className="homepage-block__label"
      >
        Adressentabel
      </span>
    </button>
  </div>
  <div
    className="homepage-block__item"
  >
    <button
      className="c-link homepage-block__link"
      onClick={[Function]}
      title="Bekijk handelsregister-tabel"
    >
      <span
        className="homepage-block__icon homepage-block__icon--hr"
      />
      <span
        className="homepage-block__label"
      >
        Handelsregister-tabel
      </span>
    </button>
  </div>
  <div
    className="homepage-block__item "
  >
    <button
      className="c-link homepage-block__link"
      onClick={[Function]}
      title="Bekijk kadaster-tabel"
    >
      <span
        className="homepage-block__icon homepage-block__icon--brk"
      />
      <span
        className="homepage-block__label"
      >
        Kadaster-tabel
      </span>
    </button>
  </div>
</HomepageBlock>
`;
